docs/v0.9/en/user-guide/operations/compaction.md

```diff
@@ -0,0 +1,134 @@
+
+For window assignment, SST files may span multiple time windows. TWCS assigns SSTs based on their maximum timestamps to ensure they are not affected by stale data. In time-series workloads, out-of-order writes are infrequent, and even when they occur, recent data's query performance is more critical than that of stale data.
+
+
+TWCS provides 4 parameters:
+- `max_active_window_runs`: max allowed sorted runs in the active window
+- `max_active_window_files`: max allowed files in the active window
+- `max_inactive_window_runs`: max allowed sorted runs in inactive windows
+- `max_inactive_window_files`: max allowed files in inactive windows
+
+Users can set different thresholds for active and inactive windows. This is important because out-of-order writes typically occur in the active window. By allowing more overlapping files in the active window, TWCS reduces write amplification during ingestion and merges all these files when the active window becomes inactive.
+
+Following diagrams show how files in an active window get compacted. 
+In A, there're two SST files `[0,3]` and `[5,6,9]` but there's only one sorted run since those two files have disjoint time ranges. 
+In B, a new SST file `[1,4]` is flushed therefore forms two sorted runs. Then `[0,3]` and `[1,4]` are merged to `[0,1,3,4]`.
+In C, a new SST file `[9,10]` is flushed, and it will be merged with `[5,6,10]` to create `[5,6,9,10]`.
+D is the final state, in two compacted files form one sorted run.
+
+![compaction-twcs-active.jpg](/compaction-twcs-active.jpg)
+
+### Specifying TWCS parameters
+Users can specify TWCS parameters mentioned above while creating tables, for example:
+
+```sql
+CREATE TABLE monitor (
+  host STRING,
+  ts TIMESTAMP DEFAULT CURRENT_TIMESTAMP() TIME INDEX,
+  cpu FLOAT64 DEFAULT 0,
+  memory FLOAT64,
+  PRIMARY KEY(host))
+WITH (
+    'compaction.type'='twcs', 
+    'compaction.twcs.max_active_window_runs'='4', 
+    'compaction.twcs.max_active_window_files'='8', 
+    'compaction.twcs.max_inactive_window_runs'='1',
+    'compaction.twcs.max_inactive_window_files'='2'
+    );
+```
+
+## Strict window compaction strategy and manual compaction
+
+Unlike TWCS, which assigns one window per SST file based on their maximum timestamps, the strict window strategy assigns SST files to **all** overlapping windows. Consequently, a single SST file may be included in multiple compaction outputs, as its name suggests. Due to its high read amplification, the strict window is not the default compaction strategy. However, it is useful when users need to manually trigger compaction to reorganize the layout of SST files—especially if an individual SST file spans a large time range that significantly slows down queries. GreptimeDB offers a simple SQL function for triggering compaction:
```

Comment:
So this read amplification refers to what happens during the compaction process? It feels necessary to clarify this. Otherwise, it might lead to the assumption that it affects queries.

Reply:
fixed in 9b50c7a7